### PR TITLE
fix(search): navigation URL path change

### PR DESCRIPTION
**Pull request by sourabpramanik**
## Description
Changed the path name in navigating URL so now after selection of the search result from the list the user will be navigated to correct page.

And removing the filter in the dashboard won't lead the user to 404. 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [X] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
This PR fixes both issues #505 and [#2423 in open-sauced/app repo](https://github.com/open-sauced/app/issues/2423)

## Mobile & Desktop Screenshots/Recordings

fix

## Steps to QA
1.go to https://hot.opensauced.pizza/
2. type rust
3. wait for the results, click on the first one to select it, then press Enter to navigate
4. try removing filter now

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [X] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
NO

### CODE REVIEW

File: src/components/Hero.tsx [hunk 1]
```diff
@@ -120,16 +105,14 @@ const Hero = () => {
...
                 </div>
 
-                {fetchedData.map(data => (
+                {fetchedData.map((data) => (
                   <Combobox.Option
                     key={data.full_name}
                     as="a"
                     className={({ active }) => (active ? "bg-gray-50" : "")}
-                    value={`https://insights.opensauced.pizza/hot/repositories/filter/${data.full_name}`}
+                    value={`https://insights.opensauced.pizza/${data.name}/dashboard/filter/${data.full_name}`}
```
Comment (jpmcb) [inline src/components/Hero.tsx:113]: We might as well change this to use the updated `app.opensauced.pizza`: we do have a DNS redirect setup, but better to not rely on that in this app code.

**Review by jpmcb — COMMENTED** [review]
Thanks for picking this up!!!! 🚀 

A few linting errors it looks like: please also address those! Thanks!!

File: src/components/Hero.tsx (hunk 1 above)
Comment (sourabpramanik) [reply src/components/Hero.tsx:113]: Understood, I will come back with linting error fixes

**Comment by nickytonline** [comment]
As soon as I type the letter `r` in the deploy preview, the screen goes blank and there are errors in the console.

CleanShot 2024-01-08 at 16 06 37

```
supabase.ts:120 {code: '42501', details: null, hint: null, message: 'permission denied for schema public'}
overrideMethod @ console.js:213
Ou @ supabase.ts:120
await in Ou (async)
(anonymous) @ Hero.tsx:50
(anonymous) @ useDidUpdate.js:28
Ki @ react-dom.production.min.js:244
ma @ react-dom.production.min.js:286
(anonymous) @ react-dom.production.min.js:282
z @ scheduler.production.min.js:13
B @ scheduler.production.min.js:14
Show 1 more frame
Show less
react-dom.production.min.js:189 TypeError: Cannot read properties of null (reading 'length')
    at bK (Hero.tsx:114:26)
    at Lf (react-dom.production.min.js:167:137)
    at Kl (react-dom.production.min.js:197:258)
    at im (react-dom.production.min.js:292:88)
    at am (react-dom.production.min.js:280:389)
    at s0 (react-dom.production.min.js:280:320)
    at ki (react-dom.production.min.js:280:180)
    at iu (react-dom.production.min.js:271:88)
    at tm (react-dom.production.min.js:268:429)
    at z (scheduler.production.min.js:13:203)

Hero.tsx:114 Uncaught TypeError: Cannot read properties of null (reading 'length')
    at bK (Hero.tsx:114:26)
    at Lf (react-dom.production.min.js:167:137)
    at Kl (react-dom.production.min.js:197:258)
    at im (react-dom.production.min.js:292:88)
    at am (react-dom.production.min.js:280:389)
    at s0 (react-dom.production.min.js:280:320)
    at ki (react-dom.production.min.js:280:180)
    at iu (react-dom.production.min.js:271:88)
    at tm (react-dom.production.min.js:268:429)
    at z (scheduler.production.min.js:13:203)
```

**Comment by sourabpramanik** [comment]
Yes @nickytonline the supabase env variables VITE_SUPABASE_URL and VITE_SUPABASE_API_KEY are not working correctly. 

I actually used the production supabase URL and API key ( from headers when you make requests ) to fix this issue 😁

**Comment by bdougie** [comment]
Sorry for letting this one go stale @sourabpramanik. We have some updates that will make this PR not as useful. 

https://github.com/open-sauced/app/issues/2961

I am updating the URL here, but let me know if you want to continue the path of improving the experience on this project.

**Comment by sourabpramanik** [comment]
> I am updating the URL here, but let me know if you want to continue the path of improving the experience on this project.

Sure thing @bdougie

**Comment by bdougie** [comment]
This PR makes this not worth pursuing as is. We would need an entirely different approach for search. 

Most like something similar to app.opensauced.pizza, where we use the GitHub API directly. 

https://github.com/open-sauced/app/blob/beta/lib/hooks/useSearchRepos.ts

**Pull request closed without merging**